shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java

```diff
@@ -202,6 +210,47 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
     }
   }
 
+  @VisibleForTesting
+  public static void writeCommandDocs(Map<String, Command> commands, String filePath)
+      throws IOException {
+    if (commands.size() == 0) {
+      return;
+    }
+    FileWriter fileWriter;
+    Closer closer = Closer.create();
+
+    try {
+      for (Map.Entry<String, Command> mt : commands.entrySet()) {
+        fileWriter = new FileWriter(PathUtils.concatPath(filePath, mt.getValue().getCommandName() + ".yml"));
+        closer.register(fileWriter);
+        fileWriter.append(mt.getValue().getDocumentation());
+        if (!mt.getValue().getOptions().getOptions().isEmpty()) {
+          fileWriter.append("options: |\n");
+        }
+        for (Option commandOpt:mt.getValue().getOptions().getOptions()){
+          if (commandOpt.getOpt() == null){
+            fileWriter.append("  - `--"+ commandOpt.getLongOpt()+ "` ");
+          }
+            else{ fileWriter.append("  - `-"+ commandOpt.getOpt()+ "` "); }
+          fileWriter.append(commandOpt.getDescription() + "\n");
+        }
+        fileWriter.append("example: |\n  ");
+        fileWriter.append(mt.getValue().getExample());
+      }
```

Comment:
This whole method is very hacky. I don't think we should write YAML like this. Can we use Jackson to marshal this class back to YAML?